### PR TITLE
Put raiden/api and raiden/transfer under mypy checks

### PR DESCRIPTION
**Pull request by pirapira**
This PR puts `raiden/api` and `raiden/transfer` under mypy checks.  The focus is still on `Address` related errors https://github.com/raiden-network/raiden/issues/3120.

When I solve all `Address` related mypy errors, the integration of the new type system https://github.com/raiden-network/raiden/issues/3051 will be much easier.

### CODE REVIEW

File: raiden/api/python.py [hunk 1]
```diff
@@ -736,7 +736,7 @@ def get_raiden_internal_events_with_timestamps(self, limit: int = None, offset:
 
     def get_blockchain_events_network(
             self,
-            registry_address: typing.PaymentNetworkID,
```
Comment (palango) [inline raiden/api/python.py]: I think that was correct before, did this cause problems?

Comment (pirapira) [reply raiden/api/python.py]: Thanks for finding I was going to modify 99% of similar things instead of fixing 1%.

Comment (pirapira) [reply raiden/api/python.py]: When I reverted this change, I had to add one type casting instead. This one makes sense to me.  https://github.com/raiden-network/raiden/pull/3144/commits/af11ca0c2d73202178cbdccbd30eaa0b42e816e6#diff-62070ecf9e8fed09dea509b5b3957664R87

**Comment by pirapira** [comment]
Thanks for the feedback.  I can choose better next time (I hope).

File: raiden/api/python.py (hunk 1 above)
Comment (palango) [reply raiden/api/python.py]: Yes, I agree. This way we have a special type in the codebase and only convert it at the boundary.

**Pull request merged**